controllers/nicclusterpolicy_controller.go

```diff
@@ -115,6 +122,39 @@ func (r *NicClusterPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	return ctrl.Result{}, nil
 }
 
+func (r *NicClusterPolicyReconciler) updateNodeLabels(cr *mellanoxv1alpha1.NicClusterPolicy,
+	infoCatalog state.InfoCatalog) error {
+	// TODO: support case if MOFED is not configured
+	nodeInfo := infoCatalog.GetNodeInfoProvider()
+	if nodeInfo == nil {
+		return errors.New("catalog does not provide node information")
+	}
+	if cr.Spec.OFEDDriver != nil {
+		attrs := nodeInfo.GetNodesAttributes(
+			nodeinfo.NewNodeLabelFilterBuilder().WithLabel(nodeinfo.NodeLabelMlnxNIC, "true").Build())
+		var pods corev1.PodList
+		appLabel := "mofed-" + attrs[0].Attributes[nodeinfo.AttrTypeOSName] + attrs[0].Attributes[nodeinfo.AttrTypeOSVer]
```

Comment:
technically you could define a new label dentifying a pod as a "mofed driver container" regardless of os and arch 
(which is used for the daemonset to own these pods in case of mixed environments which we dont support atm :))

then you would not need to fetch nodes and just use the fixed label.

eg label  `driver-pod: mofed-<version>` or similar

Reply:
Let's do it! It will simplify the code without any side effects